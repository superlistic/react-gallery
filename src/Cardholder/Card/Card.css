@import url('https://fonts.googleapis.com/css2?family=Dawning+of+a+New+Day&display=swap');
.card {
  height: 400px;
  width: 400px;
  background-color: #ffffff;
  color: var(--clr-dark);
  font-family: 'Dawning of a New Day', cursive;
  margin: 2px;
}
.card > * {
  object-fit: cover;
  height: 400px;
  width: 400px;
}
.card > p {
  font-size: calc(3rem);
  line-height: 1;
  padding: 2rem;
  text-align: center;
  justify-content: center;
}
