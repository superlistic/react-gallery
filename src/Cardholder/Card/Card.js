import React, { useState } from 'react';
import './Card.css';
import '../../main.css';
const Card = ({ image }) => {
  const [clicked, setClicked] = useState(false);
  const className = clicked ? 'card card--selected' : 'card';
  return (
    <div
      className={className}
      data-testid={`card_${image.id}`}
      onClick={() => {
        setClicked(!clicked);
      }}
    >
      <div className="card__inner">
        <div className="card__front">
          <img
            className="card__image"
            data-testid={`image_${image.id}`}
            src={image.urls.regular}
            alt={image.alt_description}
          />
        </div>
        <div className="card__back">
          <p>{image.description}</p>
        </div>
      </div>
    </div>
  );
};

export default Card;
