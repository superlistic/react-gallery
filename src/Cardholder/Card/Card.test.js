import '@testing-library/jest-dom/extend-expect';
import Card from './Card';
import React from 'react';
import renderer from 'react-test-renderer';
import { data } from '../../mockData';
const image = data.results[8];

test('renders elements', () => {
  const tree = renderer.create(<Card image={image} />).toJSON();
  expect(tree).toMatchSnapshot();
});
