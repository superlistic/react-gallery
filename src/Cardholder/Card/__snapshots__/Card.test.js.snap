// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders elements 1`] = `
<div
  className="card"
  data-testid="card_NodtnCsLdTE"
  onClick={[Function]}
>
  <div
    className="card__inner"
  >
    <div
      className="card__front"
    >
      <img
        alt="brown Scottish fold in brown thick-pile blanket"
        className="card__image"
        data-testid="image_NodtnCsLdTE"
        src="https://images.unsplash.com/photo-1494256997604-768d1f608cac?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjE3NTg1Mn0"
      />
    </div>
    <div
      className="card__back"
    >
      <p>
        Crouching Tiger, Hidden Dragon.
Photo of a kitten named William hiding under a plaid
      </p>
    </div>
  </div>
</div>
`;
